pkg/controller/backupentry/actuator.go

```diff
@@ -43,3 +52,38 @@ func (a *actuator) Delete(ctx context.Context, _ logr.Logger, be *extensionsv1al
 	entryName := strings.TrimPrefix(be.Name, v1beta1constants.BackupSourcePrefix+"-")
 	return util.DetermineError(storageClient.DeleteObjectsWithPrefix(ctx, be.Spec.BucketName, fmt.Sprintf("%s/", entryName)), helper.KnownCodes)
 }
+
+func (a *actuator) injectWorkloadIdentityData(ctx context.Context, be *extensionsv1alpha1.BackupEntry, data map[string][]byte) error {
+	backupEntrySecret := &corev1.Secret{}
+	if err := a.client.Get(ctx, kutil.ObjectKeyFromSecretRef(be.Spec.SecretRef), backupEntrySecret); err != nil {
+		return err
+	}
+
+	if backupEntrySecret.Labels == nil {
+		return nil
+	}
+
+	if backupEntrySecret.Labels[securityv1alpha1constants.LabelPurpose] != securityv1alpha1constants.LabelPurposeWorkloadIdentityTokenRequestor {
+		return nil
+	}
+
+	if backupEntrySecret.Labels[securityv1alpha1constants.LabelWorkloadIdentityProvider] != gcp.Type {
+		return nil
+	}
```

Comment:
I wonder if we can use a helper function such as 

```golang
func IsWorkloadIdentitySecret(secret *corev1.Secret) bool {
    if secret.Labels == nil {
        return false
    }
    return secret.Labels[securityv1alpha1constants.LabelPurpose] == securityv1alpha1constants.LabelPurposeWorkloadIdentityTokenRequestor &&
           secret.Labels[securityv1alpha1constants.LabelWorkloadIdentityProvider] == "gcp"
}
```

Such can also be provided by gardener/gardener repository and accept a type parameter as well. 

It can also be used by the cloudprovider secret ensurer https://github.com/gardener/gardener-extension-provider-gcp/blob/2e7a86312e7d84f49c4374e9da055cbb41cba8bb/pkg/webhook/cloudprovider/ensurer.go#L55-L57